### DISCUSSION TITLE
Populate database using a barcode based workflow

### DISCUSSION THREAD

**Discussion by mittkkoo** [inventree/InvenTree#5695] [Ideas]
After getting some very useful hints about EAN in #5682 and realising that most of our suppliers offer very useful stuff like individual pricing updates, descriptions and more, based on EAN of their products, I came up with the following sketch. The idea is to populate inventree's database by simply scanning unknown EAN barcodes with the mobile app. All other part details should come from the datasets of the suppliers. 

1. Mobile app scans a barcode / QR code
2. A plugin checks if this is a valid EAN code.
3. If no -> continue with the default barcode plugin, i.e. search for assigned barcodes and so on.
4. If yes -> search for this EAN in inventree's database in 2 fields: the part parameter named `EAN` and in the part description
5. If found -> show that part and **finish**.
6. If not found (**the interesting part**)-> search the EAN in all locally stored DATANORM datasets (these are ASCII files), provided by several different suppliers and updated daily.
7. If not found in a DATANORM dataset-> open the Create new part wizard with the EAN prefilled in the part parameter `EAN` and everything else blank, and **finish**.
8. If found in a DATANORM dataset-> create these objects:
   - one new part with fields extracted from DATANORM
   - 1 or more supplier parts with the following fields extracted from one or more DATANORM datasets: name, SKU, description, price, units, package, etc.
   - 1 manufacturer part, if at least the manufacturer is contained in  DATANORM and also get the MPN. Beforehand create the manufacturer, if not already available
 9. Show the newly created part and **finish**.

I want to use the Barcode mixin to start with. Do you think this workflow is possible? Is there something missing / not implemented?

**Comment by matmair** [comment]
It is possible, similar things have been done by users in the community.

**Reply by willbicks** [reply]
I'm also interested in this workflow, do you happen to have links to any of those examples of prior art?
